### PR TITLE
SUR-252 Responsive Implementation for Container component ( Responsive Flexbox )

### PR DESCRIPTION
**Pull request by JelenaTakac**
### Description

# Container

## Description

The `Container` component is a flexible layout component that can be used to create grid or flexbox-based layouts.

### containerType
- **Type:** `string`
- **Default:** `"flex"`
- **Description:** Specifies the type of container
    - `"flex"`
    - `"grid"`

### gap
- **Type:** `string`
- **Default:** `"sm"`
- **Description:** Defines the gap between grid or flex items.
    - `"xs"`
    - `"sm"`
    - `"md"`
    - `"lg"`
    - `"xl"`
    - `"2xl"`

### direction
- **Type:** `string`
- **Default:** `""`
- **Description:** Defines the direction of flex items (applicable only for flex containers).
    - `"row"`
    - `"row-reverse"`
    - `"column"`
    - `"column-reverse"`

### justify
- **Type:** `string`
- **Default:** `""`
- **Description:** Specifies how flex items are aligned along the main axis.
    - `"normal"`
    - `"start"`
    - `"end"`
    - `"center"`
    - `"between"`
    - `"around"`
    - `"evenly"`
    - `"stretch"`

### align
- **Type:** `string`
- **Default:** `""`
- **Description:** Specifies how flex items are aligned along the cross axis.
    - `"start"`
    - `"end"`
    - `"center"`
    - `"baseline"`
    - `"stretch"`

### wrap
- **Type:** `string`
- **Description:** Defines how flex items should wrap in the container. If no wrap is specified, it will automatically wrap if any column props (cols, tabCols, mCols) are provided.
    - `"nowrap"`
    - `"wrap"`
    - `"wrap-reverse"`

### cols
- **Type:** `string`
- **Default:** `""`
- **Description:** Specifies the number of columns for desktop view (applies to flexbox layouts). The column width is dynamically calculated based on the number of columns and the gap size.

### tabCols
- **Type:** `string`
- **Default:** `""`
- **Description:** Specifies the number of columns for tablet view (applies to flexbox layouts). The column width is dynamically calculated based on the number of columns and the gap size.

### mCols
- **Type:** `string`
- **Default:** `""`
- **Description:** Specifies the number of columns for mobile view (applies to flexbox layouts). The column width is dynamically calculated based on the number of columns and the gap size.

### className
- **Type:** `string`
- **Description:** Additional custom classes to be added to the container.

### children
- **Type:** `ReactNode`
- **Description:** The content to be rendered inside the container.

### extraProps
- **Type:** `object`
- **Description:** Additional props to be passed to the container's root element.

### grow
- **Type:** `number`
- **Description:** Specifies how much a flex item should grow relative to the rest of the flex items.
    - `"0"`
    - `"1"`

### shrink
- **Type:** `number`
- **Description:** Specifies how much a flex item should shrink relative to the rest of the flex items.
    - `"0"`
    - `"1"`

### order
- **Type:** `string`
- **Description:** Defines the order of the flex item in the container.
    - `"1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12"`
    - `"first"`
    - `"last"`
    - `"none"`

### alignSelf
- **Type:** `string`
- **Description:** Allows the default alignment (or the one specified by align-items) to be overridden for individual flex items.
    - `"auto"`
    - `"start"`
    - `"end"`
    - `"stretch"`
    - `"baseline"`

### Basic Example

```jsx
import { Container } from '@bsf/force-ui';

const App = () => (
    <>
        {/* Default Flex Behavior */}
        <Container justify="end" align="start" gap="xs" className="bg-orange-300 h-60">
            <Container.Item grow={1}>Item 1</Container.Item>
            <Container.Item>Item 2</Container.Item>
            <Container.Item grow={1}>Item 3</Container.Item>
        </Container>

        {/* Responsive Behavior */}
        <Container cols={5} tabCols={3} mCols={1} className="bg-pink-200 p-4 gap-0">
            <Container.Item className="bg-red-500 p-4">Item 1</Container.Item>
            <Container.Item className="bg-green-500 p-4">Item 2</Container.Item>
            <Container.Item className="bg-blue-500 p-4">Item 3</Container.Item>
            <Container.Item className="bg-yellow-500 p-4">Item 4</Container.Item>
            <Container.Item className="bg-purple-500 p-4">Item 5</Container.Item>
            <Container.Item className="bg-pink-500 p-4">Item 6</Container.Item>
        </Container>
    </>
);

export default App;
```

### Checklist:

-   [ ] My code is tested
-   [ ] My code passes the PHPCS tests
-   [ ] I've created the npm build.
-   [ ] My code follows accessibility standards 
-   [ ] My code has proper inline documentation 
-   [ ] I've included any necessary tests 
-   [ ] I've included developer documentation 
-   [ ] I've added proper labels to this pull request

### CODE REVIEW

**Review by ravindra114 — APPROVED** [review]
Tested! 
Attaching video below 

https://github.com/user-attachments/assets/ab4cf67c-e976-483c-b5cc-3bc44b6077c2

File: src/components/input/input.jsx [hunk 1]
```diff
@@ -36,19 +34,8 @@ const Input = (
...
-			if ( newValue.length > 0 ) {
-				setSelectedFile( newValue[ 0 ].name );
-			} else {
-				setSelectedFile( null );
-			}
-		} else {
-			newValue = event.target.value;
-		}
-
-		if ( ! isControlled && type !== 'file' ) {
```
Comment (vrundakansara) [inline src/components/input/input.jsx:51]: Why have we deleted this code?

**Review by vrundakansara — CHANGES_REQUESTED** [review]

File: src/components/container/container.jsx [hunk 2]
```diff
@@ -1,19 +1,42 @@
 import React from 'react';
 import { cn } from '../../utilities/functions';
```
Comment (jaieds) [inline src/components/container/container.jsx:2]:
```suggestion
import { cn } from '@/utilities/functions';
```

File: src/components/container/container.jsx [hunk 3]
```diff
@@ -1,19 +1,42 @@
...
+// Helper function to convert Tailwind gap classes to rem values
+const tailwindGapToRem = ( gapClass ) => {
+	const gapMap = {
+		'gap-2': '0.5rem',
+		'gap-4': '1rem',
+		'gap-5': '1.25rem',
+		'gap-6': '1.5rem',
+		'gap-8': '2rem',
+	};
+	return gapMap[ gapClass ] || '1rem'; // Default to 1rem if not found
```
Comment (jaieds) [inline src/components/container/container.jsx:13]:
```suggestion
	return gapMap?.[ gapClass ] ?? '1rem'; // Default to 1rem if not found
```

**Review by vrundakansara — APPROVED** [review]

File: src/components/container/container.jsx [hunk 4]
```diff
@@ -58,13 +68,65 @@ const Container = ( props ) => {
...
+		4: 'md:w-1/4',
+		5: 'md:w-1/5',
+		6: 'md:w-1/6',
+		7: 'md:w-1/7',
+		8: 'md:w-1/8',
+		9: 'md:w-1/9',
+		10: 'md:w-1/10',
+		11: 'md:w-1/11',
+		12: 'md:w-1/12',
+	}?.[ tabCols ] ?? 'w-1/2';
```
Comment (jaieds) [inline src/components/container/container.jsx:103]:
```suggestion
	}?.[ tabCols ] ?? 'md:w-1/2';
```

File: src/components/container/container.jsx [hunk 5]
```diff
@@ -58,13 +68,65 @@ const Container = ( props ) => {
...
+		4: 'lg:w-1/4',
+		5: 'lg:w-1/5',
+		6: 'lg:w-1/6',
+		7: 'lg:w-1/7',
+		8: 'lg:w-1/8',
+		9: 'lg:w-1/9',
+		10: 'lg:w-1/10',
+		11: 'lg:w-1/11',
+		12: 'lg:w-1/12',
+	}?.[ cols ] ?? 'w-1/4';
```
Comment (jaieds) [inline src/components/container/container.jsx:118]:
```suggestion
	}?.[ cols ] ?? 'lg:w-1/4';
```

**Review by jaieds — APPROVED** [review]

**Pull request merged**